### PR TITLE
PAYARA-4030 Upgrade HK2 to 2.6.1 and ASM to 7.1

### PR DESCRIPTION
**Pull request by Cousjava**

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/registration/glassfish-registration/src/main/java/com/sun/enterprise/registration/glassfish/ModuleMap.java [hunk 1]
```diff
@@ -159,7 +159,7 @@
                 put("org.glassfish.hk2.config", 100);
                 put("org.glassfish.hk2.config-types", 101);
                 put("org.glassfish.hk2.core", 102);
-                put("org.glassfish.hk2.external.asm-repackaged", 103);
+                //put("org.glassfish.hk2.external.asm-repackaged", 103);
                 //put("org.glassfish.hk2.external.bean-validator", 104);
```
Comment (cubastanley) [inline appserver/registration/glassfish-registration/src/main/java/com/sun/enterprise/registration/glassfish/ModuleMap.java]: Should commented code be removed?

File: nucleus/core/kernel/pom.xml [hunk 2]
```diff
@@ -258,7 +258,11 @@
             <groupId>org.glassfish</groupId>
             <artifactId>javax.el</artifactId>
         </dependency>
-        
+        <dependency>
```
Comment (MarkWareham) [inline nucleus/core/kernel/pom.xml]: Does this change have potential to impact test suites? - I recall a problem in testing in the past ~3 weeks to do with tiger-types

File: nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties [hunk 3]
```diff
@@ -183,14 +183,18 @@ glassfish.osgi.auto.install=\
 hk2.bundles=\
  ${com.sun.aas.installRootURI}modules/hk2-api.jar \
  ${com.sun.aas.installRootURI}modules/hk2-utils.jar \
- ${com.sun.aas.installRootURI}modules/javax.inject.jar \
+ ${com.sun.aas.installRootURI}modules/jakarta.inject.jar \
```
Comment (MarkWareham) [inline nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties]: Is this related?

File: nucleus/core/kernel/pom.xml (hunk 2 above)
Comment (Cousjava) [reply nucleus/core/kernel/pom.xml]: This what has been fixed in 2.6.1 of HK2 - it has an OSGi import of tiger-types that was unused within it and has now been removed. See eclipse-ee4j/glassfish-hk2#469.

File: nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties (hunk 3 above)
Comment (Cousjava) [reply nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties]: Yes, HK2 2.6 depends on Jakarta injection
